core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainerFactory.scala

```diff
@@ -126,6 +129,13 @@ object DockerContainerFactoryProvider extends ContainerFactoryProvider {
                                    logging: Logging,
                                    config: WhiskConfig,
                                    instanceId: InstanceId,
-                                   parameters: Map[String, Set[String]]): ContainerFactory =
-    new DockerContainerFactory(config, instanceId, parameters)(actorSystem, actorSystem.dispatcher, logging)
+                                   parameters: Map[String, Set[String]]): ContainerFactory = {
+
+    new DockerContainerFactory(config, instanceId, parameters, dockerApiFactory)(
+      actorSystem,
+      actorSystem.dispatcher,
+      logging)
+  }
+  def dockerApiFactory(actorSystem: ActorSystem, logging: Logging) =
```

Comment:
Why pull this out to a seperate function?

Reply:
wasn't sure how else to make it testable - as a mock https://github.com/apache/incubator-openwhisk/pull/3211/files#diff-16241f60cd0658f397e5c38d57963c95R63

Will take another look to see if I can simplify it a bit

Reply:
ok, removed the function; the args on DockerContainerFactory are a little out of control, but maybe this is a bit simpler